airbyte-integrations/connectors/source-google-analytics-data-api/unit_tests/test_authenticator.py

```diff
@@ -0,0 +1,56 @@
+import datetime
+
+from source_google_analytics_data_api import utils
+from source_google_analytics_data_api.authenticator import GoogleServiceKeyAuthenticator
+
+
+private_key = """
+-----BEGIN RSA PRIVATE KEY-----
+MIIBPAIBAAJBAIy64eoS8VCwNnu6+kcyvRc7w/Cw20fnZWcpftLIl33ZWdXl/Q+W
+sEQkm2RRWO2R9CGC2bJRZYEbiAabuG4T1LkCAwEAAQJBAIbr5Qv1fUZOqu2VJb58
+9qz/r6ti49jcEGwHbH/JsPQFpXWTyKdonibIblLDB4AbZdR25zEM2k2G42OErWjJ
+0wECIQD21o4uOMJtAe7GLvg3AeQrb4t9sWlCPuNNxnmKfn8PfQIhAJH0F7vSyD7D
+UwRcMht0PU65zWUZArBaI7BlpmFhgNbtAiEA1rIZ6vQtkDjtIW37MYUwm+Milgo4
+vokKljx6vM5339UCIEb1Owyvn3cUEypNgHbkfmHl5zu9exct26gI42j4tGDJAiEA
+3uFuRTUY3W5s8hxyHcATsGhWfGa4VCSjlKE+sSchksc=
+-----END RSA PRIVATE KEY-----
+"""
+
+
+def test_authenticator(mocker):
```

Comment:
nit: requests_mock is a better option to mock http requests, since is less erroneous (see https://github.com/airbytehq/airbyte/pull/18931)